jalantechnologies/rflask-boilerplate
# Feat: Add SignUp Flow

## Description
_This PR aims to add the functionality of SignUp to the Boilerplate code.
The UI can be seen in the screenshots and functionality in the video attached._

## Screenshots:

| <img width="1840" alt="Screenshot 2024-03-22 at 00 11 51" src="https://github.com/jalantechnologies/rflask-boilerplate/assets/128154648/8a5fc9a0-ef48-464e-87b1-fcb2e1ab0a94"> | <img width="1840" alt="Screenshot 2024-03-22 at 00 12 44" src="https://github.com/jalantechnologies/rflask-boilerplate/assets/128154648/b7dd2b4c-89ce-4369-ba1e-c49d09226baf"> |
| ---- | ---- |
| <img width="1840" alt="Screenshot 2024-03-22 at 00 12 37" src="https://github.com/jalantechnologies/rflask-boilerplate/assets/128154648/3fc481b4-9445-4fe5-8ecb-9d7b37185f0a"> | <img width="1840" alt="Screenshot 2024-03-22 at 00 12 24" src="https://github.com/jalantechnologies/rflask-boilerplate/assets/128154648/f315eb92-dd65-4f74-9fd7-53eb88f54010"> |

## Video:
_Video demonstrating feature of `Sign Up` flow._



https://github.com/jalantechnologies/rflask-boilerplate/assets/128154648/4e73da3d-826b-4d53-82a2-b77b71fd8e4d

_Video demonstrating all the validations of `Sign Up` page._

https://github.com/jalantechnologies/rflask-boilerplate/assets/128154648/30b2006c-b2e4-413c-bbd7-f7036afa3e6a



## Database schema changes

```
Account:
  id: str
  first_name: str
  hashed_password: str
  last_name: str
  username: str
```

## Tests
### Automated test cases added
- Modified existing Automated test cases to match DB Schema changes

### Manual test cases run
_For each manual test case, list the steps to test or reproduce the PR._

- Verified on clicking `Sign Up` button with one or more of the field/s blank (First Name, Last Name, Email, Password, Confirm Password) didnot submit the form and raises error.
- Verified validations of all Input Fields
- Verified Account Creation on Clicking `Sign Up` button when all the information is correct.
- Checked raising of error if the same Email is already created.
